core/vm/evm.go

```diff
@@ -286,6 +289,7 @@ func (evm *EVM) StaticCall(caller ContractRef, addr common.Address, input []byte
 	ret, err = run(evm, snapshot, contract, input)
 	if err != nil {
 		evm.StateDB.RevertToSnapshot(snapshot)
+		contract.SetVmError(err)
```

Comment:
There's no reason to use such a mechanism, since the error is returned anyway from this function.